### PR TITLE
feat: add filterResetToDefaultFilteredValue api

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/34318
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      click the reset button, whether to restore the default filter      |
| 🇨🇳 Chinese |    点击重置按钮的时候，是否恢复默认筛选值       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
filterXxx 太多了，我要找个时间迁移到 filter: { xxx }。

**Comment by heiyu4585** [comment]
> filterXxx 太多了，我要找个时间迁移到 filter: { xxx }。
那这个pr还继续吗

**Comment by afc163** [comment]
继续吧，不冲突。

File: components/table/__tests__/Table.filter.test.js [hunk 1]
```diff
@@ -2189,4 +2197,68 @@ describe('Table.filter', () => {
...
+    expect(wrapper.find('.ant-tree-checkbox-checked').length).toBe(1);
+    wrapper
+      .find(Checkbox)
+      .find('input')
+      .simulate('change', { target: { checked: true } });
+    expect(wrapper.find('.ant-tree-checkbox-checked').length).toBe(6);
+    wrapper.find('button.ant-btn-link').simulate('click', nativeEvent);
+    expect(wrapper.find('.ant-tree-checkbox-checked').length).toBe(0);
+
+    columnFilter.filterResetToDefaultFilteredValue = true;
```
Comment (afc163) [inline components/table/__tests__/Table.filter.test.js:2244]: 别改这个，下面直接

```
    wrapper = mount(
      createTable({
        columns: [{ ...columnFilter, filterResetToDefaultFilteredValue: true }],
      }),
    );
```

Comment (heiyu4585) [reply components/table/__tests__/Table.filter.test.js]: done

File: components/table/index.en-US.md [hunk 2]
```diff
@@ -120,6 +120,7 @@ One of the Table `columns` prop for describing the table's columns, Column has t
 | colSpan | Span of this column's title | number | - |  |
 | dataIndex | Display field of the data record, support nest path by string array | string \| string\[] | - |  |
 | defaultFilteredValue | Default filtered values | string\[] | - |  |
+| filterResetToDefaultFilteredValue | click the reset button, whether to restore the default filter | boolean | - |  |
```
Comment (afc163) [inline components/table/index.en-US.md:123]:
```suggestion
| filterResetToDefaultFilteredValue | click the reset button, whether to restore the default filter | boolean | false |  |
```

Comment (heiyu4585) [reply components/table/index.en-US.md]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**